domain/model/service/modelservice.go

```diff
@@ -43,26 +56,68 @@ type ControllerState interface {
 	GetModelState(context.Context, coremodel.UUID) (model.ModelState, error)
 }
 
+// EnvironVersionProvider defines a minimal subset of the EnvironProvider interface
+// that focuses specifically on the provider's versioning capabilities.
+type EnvironVersionProvider interface {
+	// Version returns the version of the provider. This is recorded as the
+	// environ version for each model, and used to identify which upgrade
+	// operations to run when upgrading a model's environ. Providers should
```

Comment:
I would avoid making business logic declarations such as provider should start at zero when this interface doesn't get to enforce this contract.

Also most of our providers are wrong or not implemented correctly in this regards so I think we should just drop the statements as none of this package applies and special meaning to the value of the number.

Reply:
That makes sense, thanks for pointing that out tom :)